mw-webapp/src/component/modal/Modal.tsx

```diff
@@ -1,44 +1,38 @@
-import {ReactElement, useState} from "react";
+import {ReactElement} from "react";
 import {Root as DialogRoot} from "@radix-ui/react-dialog";
 import {ModalContent} from "src/component/modal/ModalContent/ModalContent";
-import {ModalTrigger} from "src/component/modal/ModalTrigger/ModalTrigger";
 
 /**
  * Modal props
  */
 interface ModalProps {
 
   /**
-   * The element that triggers the modal.
+   * The content to display within the modal.
    */
-  trigger: ReactElement<HTMLElement>;
+  content: ReactElement<HTMLElement>;
 
   /**
-   * The content to display within the modal.
+   * Controls whether the modal is open or closed.
    */
-  content: ReactElement<HTMLElement>;
+  isOpen: boolean;
 
   /**
-   * Controls whether the model is initially open or closed.
-   * @default false
+   * Function that handles closing the modal.
    */
-  open?: boolean;
+  handleClose: (open: boolean) => void;
```

Comment:
If it used only for close, how does it possible that param open is boolean? If you close something, semantically open is always false

Reply:
Fixed